proof-systems/src/darlin/marlin/src/lib.rs

```diff
@@ -176,7 +190,18 @@ impl<G: Group, PC: PolynomialCommitment<G>> Marlin<G, PC> {
 
         let mut fs_rng = PC::RandomOracle::from_seed(fs_rng_init_seed)?;
         fs_rng.record::<G::BaseField, _>(index_pk.index_vk.get_hash())?;
-        fs_rng.record(public_input)?;
+
+        let x_poly_comm_time = start_timer!(|| "Committing to input poly");
+        let (init_comms, init_comm_rands) =
+            PC::commit_vec(pc_pk, prover_init_oracles.iter(), None).map_err(Error::from_pc_err)?;
```

Comment:
> In reality, let's compute the commitment to the input polynomial like the verifier does, i.e. using the lagrange commitments you've just computed before, skipping the FFT required to compute the input poly. If you want, you can introduce a test asserting that the commitment computed using the FFT + PC::commit function is the same as the one computed using the lagrange commitments

Is it really worth it? Even though it is true that the FFT could be skipped in this step, it is required nevertheless, because in later computations we need the coefficients of the input poly

Reply:
> Since it's only a single polynomial commitment, let's call the PC::commit (in the verifier you consider it already as a single commitment, and might be a bit misleading for someone reading the code)

I agree

Reply:
> Since it's only a single polynomial commitment, let's call the PC::commit (in the verifier you consider it already as a single commitment, and might be a bit misleading for someone reading the code)

Done